packages/amplify_storage_s3/ios/Classes/AmplifyStorageOperations.swift

```diff
@@ -141,16 +144,59 @@ public class AmplifyStorageOperations {
     }
     
-    private static func prepareError(flutterResult: @escaping FlutterResult, msg: String, localizedError: String?, recoverySuggestion: String?) {
-        var errorDetails: [String: Any] = [:]
-        var platformErrorDetails: [String: String] = [:]
-        
-        platformErrorDetails["platform"] = "iOS"
-        platformErrorDetails["localalizedError"] = localizedError
-        platformErrorDetails["recoverySuggestion"] = recoverySuggestion
-        
-        errorDetails["PLATFORM_EXCEPTIONS"] = platformErrorDetails
-        
-        flutterResult(FlutterError(code: "AmplifyException", message: msg, details: errorDetails))
+    private static func prepareError(flutterResult: @escaping FlutterResult, storageError: StorageError) {
+        let details: Dictionary<String, String> = createSerializedError(storageError: storageError);
+        logErrorContents(error: storageError)
+           ErrorUtil.postErrorToFlutterChannel(
+               result: flutterResult,
+               errorCode: "StorageException",
+               details: details)
+    }
+    
+    private static func prepareGenericError(flutterResult: @escaping FlutterResult, error: Error) {
+        logErrorContents(error: error)
+        var message: String = ErrorMessages.missingExceptionMessage
+        var recoverySuggestion: String = ErrorMessages.missingRecoverySuggestion
+        var underlyingError: Error? = nil
+
+        if (error is AmplifyFlutterValidationException) {
+            let e = error as! AmplifyFlutterValidationException
+            message = e.errorDescription
+            recoverySuggestion = e.recoverySuggestion
+            underlyingError = e.underlyingError
+        }
+        let serializedErrror = createSerializedError(
+            message: message,
+            recoverySuggestion: recoverySuggestion,
+            underlyingError: underlyingError.debugDescription)
+
+        ErrorUtil.postErrorToFlutterChannel(
+            result: flutterResult,
+            errorCode: "StorageException",
+            details: serializedErrror)
+            
+
+    }
+
+   static func createSerializedError(storageError: StorageError)  -> Dictionary<String, String> {
+        var serializedException: Dictionary<String, String> = [:]
+        serializedException["message"] = storageError.errorDescription
+        serializedException["recoverySuggestion"] = storageError.recoverySuggestion
+        serializedException["underlyingException"] = storageError.underlyingError.debugDescription
+        return serializedException
+    }
+
+    static func createSerializedError(message: String, recoverySuggestion: String, underlyingError: String)  -> Dictionary<String, String> {
+        var serializedException: Dictionary<String, String> = [:]
+        serializedException["message"] = message
+        serializedException["recoverySuggestion"] = recoverySuggestion
+        serializedException["underlyingException"] = underlyingError
```

Comment:
We seem to be doing `underlyingError.debugDescription` above but directly assigning the `underlyingError` here?

Reply:
In the first method, we always have a StorageError, which as an extension of AmplifyError has a debugDescription.  In this second method, we are just using the string that is passed in (which will either be "AmplifyFlutterValidationException" or the localizedDescription.

Reply:
Fixed this as we discussed... in Kotline, we can simply do `e.toString()`, and in swift we are using a custom error with it's own implementation of `LocalizedError `.